q2_quality_control/_threshold_graph/_visualizer.py

```diff
@@ -41,15 +43,48 @@ def _check_inputs(**kwargs):
 TEMPLATES = pkg_resources.resource_filename(
     'q2_quality_control._threshold_graph', 'assets')
 
+#generates seqeunce table and fasta files for each assignment
+def write_table_fastas(output_dir, dest, display_sequences, sequences, df, desig, decontam_scores, read_nums, table):
+    _blast_url_template = ("http://www.ncbi.nlm.nih.gov/BLAST/Blast.cgi?"
+                           "ALIGNMENT_VIEW=Pairwise&PROGRAM=blastn&DATABASE"
+                           "=nt&CMD=Put&QUERY=%s")
+    with open(os.path.join(output_dir, dest), 'w') as fh:
+        for index, row in df.iterrows():
+            fh.write(format_fasta(row['id'], row['Sequence']))
+            str_seq = str(row['Sequence'])
+            display_sequences.add(row['id'])
+            sequences[row['id']] \
+                = {'url': _blast_url_template % str_seq,
+                   'seq': str_seq,
+                   'contam_or_naw': desig,
+                   'p_val': decontam_scores.loc[row['id'], 'p'],
+                   'read_nums': read_nums.loc[row['id']],
+                   'prevalence': (
+                           table[row['id']] != 0).sum()}
+    return [display_sequences, sequences]
 
-def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
-                       table: pd.DataFrame, threshold: float = 0.1,
+#formats fasta files appripriatly
+def format_fasta(header, sequence):
+    return f">{header}\n{sequence}\n"
+
+#main algorithm
+def decontam_score_viz(output_dir, decontam_scores: pd.DataFrame,
+                       table: pd.DataFrame,
+                       rep_seqs: pd.Series = None,
+                       threshold: float = 0.1,
                        weighted: bool = True, bin_size: float = 0.02):
+
+    #checks inputs
```

Comment:
What is this doing?

Reply:
This is checking to make that the inputs are appropriate. There is crossover with what Qiime2 checks however it also checking that the threshold  value is between 0 and 1. Should this be checked on the Qiime2 framework side of things? Or should I just raise a valueerror if it is out of bounds? Thanks!